sklearn/covariance/graph_lasso_.py

```diff
@@ -233,6 +233,8 @@ def graphical_lasso(emp_cov, alpha, cov_init=None, mode='cd', tol=1e-4,
                             alpha_min=alpha / (n_features - 1), copy_Gram=True,
                             eps=eps, method='lars', return_path=False)
                 # Update the precision matrix
+                if not np.isfinite(coefs).all():
```

Comment:
Definitely not an expert on this. Is there a feeling on how likely/unlikely this check (inside a double loop) could affect performance in the well-conditioned case?

Reply:
That's a valid concern :-/

Reply:
I'm pretty sure it would be fine to pull it out one level. I tried to fail early but maybe checking less often is more important.